### PR TITLE
[cmake] fix condition for apple clang

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
# Description

Cmake identifies Apple clang as "AppleClang"

Fixes # (github issue)

# Checklist

## General

- [ ] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally for each commit?
- [ ] Have you formatted the code using clang-format?

## Performance improvements

- [ ] Have you submitted performance data that demonstrates performance improvements?

### New features

- [ ] Have you published an RFC for the new feature?
- [ ] Was the RFC approved?
- [ ] Have you added relevant tests?

### Bug fixes

- [ ] Have you included information on how to reproduce the issue (either in a github issue or in this PR)?
- [ ] Have you added relevant regression tests?

## [RFC](https://github.com/oneapi-src/oneDNN/tree/rfcs) PR

- [ ] Does RFC document follow the [template](https://github.com/oneapi-src/oneDNN/blob/rfcs/rfcs/template.md#onednn-design-document-rfc)?
- [ ] Have you added a link to the rendered document?

### CODE REVIEW

**Comment by vpirogov** [comment]
@ilya-lavrenov, CI results indicate that this is not the case for configuration we are testing. Which CMake version did you try?

**Comment by ilya-lavrenov** [comment]
We use 3.13 as minimum required.
Do you really need to support cmake 2.8.x ?

**Comment by densamoilov** [comment]
> Do you really need to support cmake 2.8.x ?

Good question, we may have to bump the minimum required CMake version to at least 3.5 as the [older versions have been deprecated](https://cmake.org/cmake/help/v3.29/release/3.27.html#deprecated-and-removed-features) in CMake 3.27.

**Comment by vpirogov** [comment]
> Good question, we may have to bump the minimum required CMake version to at least 3.5 as the [older versions have been deprecated](https://cmake.org/cmake/help/v3.29/release/3.27.html#deprecated-and-removed-features) in CMake 3.27.

We are using CMake 2.8.12 as minimal supported version to ensure RHEL 7 compatibility. As RHEL 7 is now out of production support we could bump it. RHEL8 has 3.11.4 and Ubuntu 20.04 has 3.16.3.

File: cmake/OpenMP.cmake [hunk 1]
```diff
@@ -23,7 +23,7 @@ endif()
 set(OpenMP_cmake_included true)
 include("cmake/Threading.cmake")
 
-if (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+if (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
```
Comment (spalicki) [inline cmake/OpenMP.cmake:26]: Currently this check applied to all Clangs on macOS, after the change only to Apple provided Clang. If we want to retain the old logic, then `MATCHES` would be a better option like: `if (APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")`. It should catch both Clangs.

**Comment by vpirogov** [comment]
Covered by #2225.

**Pull request closed without merging**